src/windows/main_window.py

```diff
@@ -2058,17 +2058,27 @@ def removeSelection(self, item_id, item_type):
     def save_settings(self):
         s = settings.get_settings()
 
-        # Save window state and geometry (saves toolbar and dock locations)
+        # Save window state and geometry (saves toolbar, dock locations and frozen UI state)
         s.set('window_state_v2', qt_types.bytes_to_str(self.saveState()))
         s.set('window_geometry_v2', qt_types.bytes_to_str(self.saveGeometry()))
+        if self.actionUn_Freeze_View.isVisible():
+            docks_frozen = 'True'
+        else:
+            docks_frozen = 'False'
+        s.set('docks_frozen', docks_frozen)
```

Comment:
I'm not crazy about using the menu-item visibility to control the setting variable — why not just update it during `freezeDocks()` and `unfreezeDocks()`? e.g:
```python
    def freezeDocks(self):
        """ Freeze all dockable widgets on the main screen (no float, moving, or closing) """
        s = settings.get_settings()
        s.set('docks_frozen', True)
        for dock in self.getDocks():
            dock.setFeatures(QDockWidget.NoDockWidgetFeatures)
```
and similar for `unFreezeDocks()`. Then nothing special needs to be done for `docks_frozen` during `save_settings()`.

Reply:
(Alternately, the settings update could be in the triggers, instead. There doesn't seem to be any strict pattern in the code for where that's done, and it shouldn't make a difference one way or the other.)

Reply:
It's not much better. You setting value on startup, me - on close. I have one `if`, you have to read all keys twice instead (on change and on close). Remember, the change triggers on startup.

Anyway, I will add new commit to reflect this changes. Do you have better solution? Global variable or so?

Reply:
If you want to avoid the extra settings lookup during load, then like I said the updates can go in the triggers instead of `freezeDocks()` / `unFreezeDocks()`. Then it'll only happen when the user changes the freeze state, regardless whether the saved value is true or false.